src/acl/Acl.cc

```diff
@@ -328,13 +330,27 @@ Acl::Node::ParseNamed(ConfigParser &parser, Node ** const head, const SBuf &acln
                A->cfgline);
     }
 
-    // add to the global list for searching explicit ACLs by name
-    assert(head && *head == Config.aclList);
-    A->next = *head;
-    *head = A;
+    const auto insertion = namedRules.emplace(A->name, A);
+    Assure(insertion.second); // FindByName() above checked that A is a new ACL
+}
 
-    // register for centralized cleanup
-    aclRegister(A);
+void
+Acl::DumpNamedRules(std::ostream &os, const char * const directiveName, NamedRules * const namedRules)
+{
+    if (namedRules) {
+        for (const auto &nameAndAcl: *namedRules) {
+            debugs(3, 3, directiveName << ' ' << nameAndAcl.first);
+            nameAndAcl.second->dumpWhole(directiveName, os);
+        }
+    }
+}
+
+void
+Acl::FreeNamedRules(NamedRules ** const namedRules)
+{
+    assert(namedRules);
+    delete *namedRules;
+    *namedRules = nullptr;
```

Comment:
The "when" precondition for this change cannot be satisfied as detailed in https://github.com/squid-cache/squid/pull/1826#discussion_r1628354287 thread.

I believe it is _possible_ to avoid manual memory management in this and similar cases (and even tried to implement that) but doing so requires heroic efforts as detailed at https://github.com/squid-cache/squid/pull/1826#discussion_r1619152043. This PR does not make things worse as far as memory management of these Config data members is concerned -- it preserves (problematic but bearable) status quo; it should not be responsible for improving that code.